locust/contrib/fasthttp.py

```diff
@@ -0,0 +1,383 @@
+
+if six.PY2:
+    from cookielib import CookieJar
+    class ConnectionRefusedError(Exception):
+        # ConnectionRefusedError doesn't exist in python 2, so we'll 
+        # define a dummy class to avoid a NameError
+        pass
+    str = unicode
+else:
+    from http.cookiejar import CookieJar
+
+from gevent.timeout import Timeout
+from geventhttpclient.useragent import UserAgent, CompatRequest, CompatResponse, ConnectionError
+from geventhttpclient.response import HTTPConnectionClosed
+
+from locust import events
+from locust.core import Locust
+from locust.exception import LocustError, CatchResponseError, ResponseError
+
+
+# Monkey patch geventhttpclient.useragent.CompatRequest so that Cookiejar works with Python >= 3.3
+# More info: https://github.com/requests/requests/pull/871
+CompatRequest.unverifiable = False
+
+# Regexp for checking if an absolute URL was specified
+absolute_http_url_regexp = re.compile(r"^https?://", re.I)
+
+# List of exceptions that can be raised by geventhttpclient when sending an HTTP request, 
+# and that should result in a Locust failure
+FAILURE_EXCEPTIONS = (ConnectionError, ConnectionRefusedError, socket.error, \
+                      SSLError, Timeout, HTTPConnectionClosed)
+
+
+def _construct_basic_auth_str(username, password):
+    """Construct Authorization header value to be used in HTTP Basic Auth"""
+    if isinstance(username, str):
+        username = username.encode('latin1')
+    if isinstance(password, str):
+        password = password.encode('latin1')
+    return 'Basic ' + b64encode(b':'.join((username, password))).strip().decode("ascii")
+
+
+class FastHttpLocust(Locust):
+    """
+    Represents an HTTP "user" which is to be hatched and attack the system that is to be load tested.
+    
+    The behaviour of this user is defined by the task_set attribute, which should point to a 
+    :py:class:`TaskSet <locust.core.TaskSet>` class.
+    
+    This class creates a *client* attribute on instantiation which is an HTTP client with support 
+    for keeping a user session between requests.
+    """
+    
+    client = None
+    """
+    Instance of HttpSession that is created upon instantiation of Locust. 
+    The client support cookies, and therefore keeps the session between HTTP requests.
+    """
+    
+    def __init__(self):
+        super(FastHttpLocust, self).__init__()
+        if self.host is None:
+            raise LocustError("You must specify the base host. Either in the host attribute in the Locust class, or on the command line using the --host option.")
+        if not re.match(r"^https?://[^/]+$", self.host, re.I):
+            raise LocustError("Invalid host (`%s`). The specified host string must be a base URL without a trailing slash. E.g. http://example.org" % self.host)
+        
+        self.client = FastHttpSession(base_url=self.host)
+
+
+class FastHttpSession(object):
+    auth_header = None
+    
+    def __init__(self, base_url):
```

Comment:
```suggestion
     def __init__(self, base_url):
	        self.base_url = base_url
	        self.cookiejar = CookieJar()
	        self.client = LocustUserAgent(max_retries=1, cookiejar=self.cookiejar, **kwargs)
```

Allow configuration of UserAgent and HTTPClient by accepting and forwarding on **kwargs.